src/Datadog.Trace/ExtensionMethods/StringExtensions.cs

```
@@ -46,5 +61,25 @@ public static string TrimEnd(this string value, string suffix, StringComparison
                     return null;
             }
         }
+
+        /// <summary>
+        /// Truncates a <see cref="string"/> to 1024 characters, if necessary.
+        /// </summary>
+        /// <param name="value">The string to convert.</param>
+        /// <returns>The truncated string</returns>
+        public static string Truncate(this string value)
+        {
+          return value.Length > 1024 ? value.Substring(0, 1024) : value;
```

Decide what to do if `value` is `null`. let the exception be raised or return a place holder?

---

Following the pattern established by the other extension methods.